src/components/rockets/Rockets.js

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import Rocket from './Rocket';
+import styles from './Rockets.module.scss';
+import { loadRockets } from '../../redux/rockets/rockets';
+import { getRockets, getLoading } from './selectors';
+
+const Rockets = () => {
+  const dispatch = useDispatch();
+  const rockets = useSelector(getRockets);
+  const loading = useSelector(getLoading);
+  useEffect(() => rockets.length === 0 && dispatch(loadRockets()), []);
+
+  const loadingMessage = <div> Loading rockets...</div>;
+  const content = (rockets) => (
+    rockets.map((rocket) => <Rocket key={rocket.id} rocket={rocket} />)
+  );
+
+  return ((!loading)) ? (
+    <div className={styles.rockets_div}>
+      {content(rockets)}
+    </div>
```

Comment:
- [ ] When fetching data from an API, the response might get delayed or it might return an error, kindly update this component to conditionally render `Rockets` and you may as well display a message if an error occurred.

Reply:
Hi @omar-labana 
This is a part from the logic of the code block: 
` return ((!loading)) ? (
    <div className={styles.rockets_div}>
      {content(rockets)}
    </div>
  ) : loadingMessage;`
  If the response might get delayed or it might return an error, I will display the **loadingMessage**  👍🏼

Reply:
Apologies for this mistake, but it was highlighted for me that it should be within the `return` function inside the components it self.

Reply:
That's fine, Thank you for review though 👍🏼 😃